source/extensions/health_checkers/http/health_checker_impl.cc

```diff
@@ -81,6 +81,12 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(const Cluster& cluster,
     service_name_matcher_.emplace(config.http_health_check().service_name_matcher());
   }
 
+  if (!Http::HeaderUtility::headerValueIsValid(path_) ||
```

Comment:
For path validation, and because it seems `headerValueIsValid` is using QUICHE's function, why not use the [`IsValidPath` function](https://github.com/google/quiche/blob/a4c007178d479e7cc301a2bae284f41c87039448/quiche/http2/adapter/header_validator.h#L44) in QUICHE?